src/apps/amp/ethereum/amp.farm.contract-position-fetcher.ts

```diff
@@ -63,30 +81,42 @@ export class EthereumAmpFarmContractPositionFetcher extends SingleStakingFarmTem
       });
   }
 
-  getContract(address: string): AmpStaking {
-    return this.contractFactory.ampStaking({ address, network: this.network });
-  }
-
-  async getFarmDefinitions(): Promise<SingleStakingFarmDefinition[]> {
-    return FARMS;
-  }
-
-  async getRewardRates() {
-    return [0];
-  }
-
-  async getStakedTokenBalance({ address }: GetTokenBalancesParams<AmpStaking>) {
-    // We rely on the API to get staked and claimable balances
-    const { supplyTotal, rewardTotal } = await this.getApiData(address);
-    return new BigNumber(supplyTotal).plus(rewardTotal).toFixed(0);
+  async getTokenBalancesPerPosition({ address }: GetTokenBalancesParams<AmpStaking>): Promise<BigNumberish[]> {
+    const { supplyTotal, rewardTotal } = await this.getAddressBalances(address);
+    return rewardTotal > supplyTotal ? [] : [supplyTotal, rewardTotal];
   }
 
-  async getRewardTokenBalances({ address }: GetTokenBalancesParams<AmpStaking>) {
-    const { rewardTotal } = await this.getApiData(address);
-    return Number(rewardTotal).toFixed(0);
+  async getPoolApys() {
+    const capacityResponse = axios.create({
```

Comment:
I suggest you add a @Cache decorator reducing the number of external calls the app integration does to your API

Reply:
Going to add the @Cache decorators to all our api calls.